pallets/capacity/src/types.rs

```diff
@@ -361,11 +359,33 @@ impl<T: Config> ProviderBoostHistory<T> {
 		Some(self.count())
 	}
 
-	/// Return how much is staked for the given era.  If there is no entry for that era, return None.
-	pub fn get_staking_amount_for_era(&self, reward_era: &T::RewardEra) -> Option<&BalanceOf<T>> {
+	/// A wrapper for the key/value retrieval of the BoundedBTreeMap.
+	pub(crate) fn get_entry_for_era(&self, reward_era: &T::RewardEra) -> Option<&BalanceOf<T>> {
 		self.0.get(reward_era)
 	}
 
+	/// Returns how much was staked during the given era, even if there is no explicit entry for that era.
+	/// If there is no history entry for `reward_era`, returns the next earliest entry's staking balance.
+	///
+	/// Note there is no sense of what the current era is; subsequent calls could return a different result
+	/// if 'reward_era' is the current era and there has been a boost or unstake.
+	pub(crate) fn get_amount_staked_for_era(&self, reward_era: &T::RewardEra) -> BalanceOf<T> {
```

Comment:
nit: so based on my understanding, if the reward_era exists it would return the balance for that but if it doesn't it would return the sum of all balances before that. But this seems to be confusing inside the same function.

Reply:
It will not return the sum of all balances. It will return whatever is the nearest previous balance if it doesn't have the reward_era key.  If you look in the test for this function it will show how it should work.

However you helped find a bug in that I forgot to implement an important detail for `list_unclaimed_rewards`. A booster should be eligible for rewards only for amounts staked during an entire era. Otherwise you could stake late in an era, unstake early the next era, and still earn full rewards. 

I've fixed this issue, updated the test, and I've also renamed a returned field from `staked_amount` to `eligible_amount` which I hope is more clear.  Please have a look at the test `check_for_unclaimed_rewards_has_eligible_rewards`  and maybe that will help understanding it together.